backend/store/services.go

```diff
@@ -31,10 +35,10 @@ func (store *Store) FindOrCreateService(ctx context.Context, project model.Proje
 			service.ProcessDescription = &val
 		}
 
-		err := store.db.Where(&model.Service{
-			ProjectID: project.ID,
-			Name:      name,
-		}).FirstOrCreate(&service).Error
+		err := store.db.Clauses(clause.OnConflict{
+			Columns:   []clause.Column{{Name: "name"}, {Name: "project_id"}},
```

Comment:
awesome, and confirmed that this is utilizing the existing `idx_project_id_name` to ensure uniqueness